src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2087,6 +2109,22 @@ private MethodSymbol InferMethod(BoundCall node, MethodSymbol method, ImmutableA
                 return definition.Construct(result.InferredTypeArguments);
             }
             return method;
+
+            BoundExpression makePlaceholder(TypeSymbolWithAnnotations argumentType, BoundExpression argument)
+            {
+                if (argumentType is null)
+                {
+                    return argument;
+                }
+
+                if (argument is BoundLocal local && local.IsDeclaration == IsDeclaration.WithInferredType)
+                {
+                    // 'out var' doesn't contribute to inference
+                    return new BoundValuePlaceholder(argument.Syntax, isNullable: null, type: null);
+                }
+
+                return new BoundValuePlaceholder(argument.Syntax, argumentType.IsNullable, argumentType.TypeSymbol);
+            }
         }
 
         private void ReplayReadsAndWrites(LocalFunctionSymbol localFunc,
```

Comment:
>ReplayReadsAndWrites [](start = 21, length = 20)

I note this method isn't implemented. Is there a planned strategy for handling local functions (i.e. replaying based on the nullability of the variables at the call site)?

Reply:
I'll let @cston extend/re-purpose https://github.com/dotnet/roslyn/issues/26639 (issue for local functions and nullability)

Reply:
Logged #27233 for local functions.